sdk/sync/common.go

```diff
@@ -69,12 +70,12 @@ func redisClient(runenv *runtime.RunEnv) (client *redis.Client, err error) {
 	client = redis.NewClient(opts)
 
 	// PING redis to make sure we're alive.
-	return client, client.Ping().Err()
+	return client, client.WithContext(ctx).Ping().Err()
 }
 
 // MustWatcherWriter proxies to WatcherWriter, panicking if an error occurs.
-func MustWatcherWriter(runenv *runtime.RunEnv) (*Watcher, *Writer) {
-	watcher, writer, err := WatcherWriter(runenv)
+func MustWatcherWriter(ctx context.Context, runenv *runtime.RunEnv) (*Watcher, *Writer) {
```

Comment:
note: this context is only used for _connecting_ to redis. Every request gets its own context.

Reply:
Would be good to note that in the godocs.